src/rpp/rpp/operators/fwd.hpp

```diff
@@ -33,6 +33,10 @@ template<typename Fn>
     requires (!utils::is_not_template_callable<Fn> || !std::same_as<void, std::invoke_result_t<Fn, utils::convertible_to_any>>)
 auto map(Fn&& callable);
 
+template<typename Fn>
+    requires rpp::constraint::observable<std::invoke_result_t<Fn, utils::convertible_to_any>>
```

Comment:
Add check over !utils::is_not_template_callable<Fn> as in map. It needed when passed function is purely template. For example `[](auto v) {return rpp::source::just(v); }` wouldn't satisfy your requirement

Reply:
done, added a test too